src/aura/CRLP_FilterGroup/CRLP_FilterGroupHelper.js

```diff
@@ -153,26 +163,129 @@
+     * unique jobId and the Filter_Group__mdt.DeveloperName value to recursively call an apex controller method to determine
+     * when the deployment has completed (by looking in a custom settings object). If it completes, the final recordId
+     * is returned. Otherwise an error message to render is returned. If the return is null, then it calls this method
+     * again to wait another second an try again.
+     * @param jobId The returns CMT deployment job id to query status for
+     * @param recordName Unique record name value (that was just inserted/updated) to query for.
+     */
+    pollForDeploymentStatus : function(cmp, jobId, recordName, counter) {
+        var helper=this;
+        var maxPollingRetryCount = 30;
+        var poller = window.setTimeout(
+            $A.getCallback(function() {
+                counter++;
+                console.log('setTimeout(' + jobId + ',' + recordName + '):' + counter);
+                var action = cmp.get("c.getDeploymentStatus");
+                action.setParams({jobId: jobId, recordName: recordName, objectType: 'Filter'});
+                action.setCallback(this, function (response) {
+                    console.log('getDeploymentStatus.callback');
+                    var state = response.getState();
+                    if (state === "SUCCESS") {
+                        // Response will be a serialized deployResult wrapper class
+                        var deployResult = JSON.parse(response.getReturnValue());
+                        console.log('deployResult=' + deployResult);
+                        // if there is a record id response
+                        if (deployResult && deployResult.completed === true && deployResult.filterGroupItem) {
+                            window.clearTimeout(poller);
+                            helper.toggleSpinner(cmp, false);
+                            helper.showToast(cmp, 'success', cmp.get("v.labels.filtersSaveProgress"), cmp.get("v.labels.filtersSaveSuccess"));
+
+                            var model = helper.restructureResponse(deployResult.filterGroupItem);
+
+                            // Save the inserted/updated record id
+                            cmp.set("v.activeFilterGroup", model.filterGroup);
+                            cmp.set("v.activeFilterGroupId", model.filterGroup.recordId);
+
+                            // for a new record, copy the activeFilterGroup map to the cachedFilterGroup map
+                            if (cmp.get("v.cachedFilterGroup") && cmp.get("v.cachedFilterGroup.recordName")) {
+                                var activeFilterGroup = cmp.get("v.activeFilterGroup");
+                                var cachedFilterGroup = cmp.get("v.cachedFilterGroup");
+                                for (var key in activeFilterGroup) {
+                                    if (activeFilterGroup.hasOwnProperty(key)) {
```

Comment:
I noticed this in rollups too, but is there value in checking for all keys instead of straight replacing the cachedFilterGroup with activeFilterGroup? My questions is whether there may be leftover properties from the cachedFilterGroup, and why we would want/need them using this method.

Reply:
Good question. Does replacing the object directly create a reference clone of the original object? Trying to remember why I'm replacing each property and that's the only thing that comes to mind.